### DISCUSSION TITLE
组网失败

### DISCUSSION THREAD

**Discussion by xiaosa2995** [opengnb/opengnb#29] [Q&A]
<img width="564" alt="image" src="https://github.com/gnbdev/opengnb/assets/30397377/a90cfbb2-2c98-496c-b1ef-45c81fe74ba7">

<img width="988" alt="image" src="https://github.com/gnbdev/opengnb/assets/30397377/438ef9ec-7b82-4f71-8406-71add51e7dce">
<img width="949" alt="image" src="https://github.com/gnbdev/opengnb/assets/30397377/204044f5-e22f-4ca2-9b51-fd75b52d69c7">

服务器上启用9993端口，电脑上这样执行，偶尔会组网成功成功率很低是什么情况(1001：手机热点，1002：公司wifi)

**Comment by gnbdev** [comment]
要明确一点：p2p的建立在复杂的网络环境下不保证100%可以成功；根据你提供的信息，目前可以进一步提升p2p建立的途径可以是
1、使用多个 index node 以帮助节点看到对端节点的其他地址和端口，不建议使用 lite mode，这样便于添加更多的index node。
2、多部署几个节点，使得 Unified Forwarding 发挥作用。

https://github.com/gnbdev/opengnb/blob/master/docs/gnb_user_manual_cn.md
